packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx

```diff
@@ -0,0 +1,1130 @@
+        // we know that the user is typing with keypresses happening
+        // within the timeout of each other so remove the clearing task
+        // and continue building the pending value with the udpated value
+        if (this._lastReadOnlyAutoCompleteChangeTimeoutId > 0) {
+          this._async.clearTimeout(this._lastReadOnlyAutoCompleteChangeTimeoutId);
+          this._lastReadOnlyAutoCompleteChangeTimeoutId = -1;
+          updatedValue = currentPendingValue + updatedValue;
+        }
+
+        let originalUpdatedValue: string = updatedValue;
+        updatedValue = updatedValue.toLocaleLowerCase();
+
+        // If autoComplete is on, attempt to find a match where the text of an option starts with the updated value
+        let items = currentOptions.map((item, index) => { return { ...item, index }; }).filter((option) => option.itemType !== DropdownMenuItemType.Header && option.itemType !== DropdownMenuItemType.Divider).filter((option) => option.text.toLocaleLowerCase().indexOf(updatedValue) === 0);
+
+        // If we found a match, udpdate the state
+        if (items.length > 0) {
+          this._setPendingInfo(originalUpdatedValue, items[0].index, items[0].text);
+
+          // Schedule a timeout to clear the pending value after the timeout span
+          this._lastReadOnlyAutoCompleteChangeTimeoutId = this._async.setTimeout(() => { this._lastReadOnlyAutoCompleteChangeTimeoutId = -1; }, this._readOnlyPendingAutoCompleteTimeout);
+          return;
+        }
+      }
+    }
+
+    // If we get here, either autoComplete is on or we did not find a match with autoComplete on.
+    // Remeber we are not allowing freeform, so at this point, if we have a pending valid value index
+    // use that; otherwise use the selectedIndex
+    let index = currentPendingValueValidIndex >= 0 ? currentPendingValueValidIndex : selectedIndex;
+
+    // Since we are not allowing freeform, we need to
+    // set both the pending and suggested values/index
+    // to allow us to select all content in the input to
+    // give the illusion that we are readonly (e.g. freeform off)
+    this._setPendingInfoFromIndex(index);
+  }
+
+  /**
+   * Walk along the options starting at the index, stepping by the delta (positive or negative)
+   * looking for the next valid selectable index (e.g. skipping headings and dividers)
+   * @param index - the index to get the next selectable index from
+   * @param delta - optional delta to step by when finding the next index, defaults to 0
+   * @returns {number} - the next valid selectable index. If the new index is outside of the bounds,
+   * it will snap to the edge of the options array. If delta == 0 and the given index is not selectable
+   */
+  private _getNextSelectableIndex(index: number, searchDirection: SearchDirection): number {
+    let { currentOptions } = this.state;
+
+    let newIndex = index + searchDirection;
+
+    newIndex = Math.max(0, Math.min(currentOptions.length - 1, newIndex));
```

Comment:
No, if you reach one of the ends of the array and are still at an option which shouldn't get focus, we won't make another call. Also, if the caller didn't tell us which way to move we won't try